mobile/src/people/people-room.tsx

```diff
@@ -25,13 +25,39 @@ const mapDispatchToProps = (dispatch: Dispatch<any>): EmployeesListDispatchProps
     onItemClicked: (employee: Employee) => dispatch( openEmployeeDetailsAction(employee))
 });
 
-export class PeopleRoomImpl extends React.Component<PeopleRoomProps & EmployeesListDispatchProps> {  
+export class PeopleRoomImpl extends React.Component<PeopleRoomProps & EmployeesListDispatchProps> {
+    public shouldComponentUpdate(nextProps: PeopleRoomProps & EmployeesListDispatchProps) {
+        if (this.props.onItemClicked !== nextProps.onItemClicked
+            || this.props.userEmployee !== nextProps.userEmployee
+        ) {
+            return true;
+        }
+
+        const predicate = (employee: Employee) => {
+            return this.props.userEmployee && employee.roomNumber === this.props.userEmployee.roomNumber;
+        };
+
+        const nextEmployees = nextProps.employees.employeesById.toArray().filter(predicate);
+
+        for (const employee of nextEmployees) {
+            const employeeId = employee.employeeId;
+            const currentEmployeeRef = this.props.employees.employeesById.get(employeeId);
+            const nextEmployeeRef = nextProps.employees.employeesById.get(employeeId);            
```

Comment:
no point in this statement, you already have this value as 'employee'

Reply:
Furthermore, you can, perhaps, put a predicate in your MapToProps function. Applies both here in and in department